tests/lib/c_lib/src/main.c

```diff
@@ -319,6 +322,321 @@ void test_bsearch(void)
+{
+	int ch = 'a', sh = 'A';
+
+	zassert_false(isupper(ch), "isupper 'a'");
+	zassert_true(isupper(sh), "isupper 'A'");
+}
+
+/**
+ *
+ * @brief Test isxdigit function
+ *
+ */
+void test_isxdigit(void)
+{
+	int ch1 = 'e', ch2 = 'z';
+
+	zassert_true(isxdigit(ch1), "isxdigit 'e'");
+	zassert_false(isxdigit(ch2), "isxdigit 'z'");
+}
+
+/**
+ *
+ * @brief Test menchr function
+ *
+ */
+static const char str[] = "testfunction";
+void test_memchr(void)
+{
+	int ch1 = 'e', ch2 = 'z';
+	char *ret;
+
+	ret = memchr(str, ch1, strlen(str));
+	zassert_not_null(ret, "memchr 'testfunction' 'e' ");
+
+	ret = memchr(str, ch2, strlen(str));
+	zassert_is_null(ret, "memchr 'testfunction' 'z'");
+}
+
+/**
+ *
+ * @brief Test memcpy function
+ *
+ */
+void test_memcpy(void)
+{
+	char arr[10];
+	char str[] = "12345";
+
+	memcpy(arr, str, sizeof(str));
+	zassert_true((memcmp(str, arr, sizeof(str))) == 0,
+		     "memcpy failed");
+}
+
+/**
+ *
+ * @brief test memmove function
+ *
+ */
+void test_memmove(void)
+{
+	char str1[7] = "aabbcc";
+	char str2[7] = "aaaabb";
+
+	zassert_not_equal(memmove(str1 + 2, str1, 4), NULL, "memmove error");
+	zassert_true(memcmp(str1, str2, 7) == 0,
+		     "memmove failed");
+}
+
+/**
+ *
+ * @brief test strcat function
+ *
+ */
+void test_strcat(void)
+{
+	char str[20] = "hello";
+	char src[20] = "world";
+
+	zassert_not_null(strcat(str, src), "strcat false");
+	zassert_true(strcmp(str, "helloworld") == 0, "test strcat failed");
+}
+
+/**
+ *
+ * @brief test strcspn function
+ *
+ */
+void test_strcspn(void)
+{
+	char *str1 = "aabbcc";
+	char *str2 = "b";
+	char *str3 = "d";
+	int ret;
+
+	ret = strcspn(str1, str2);
+	zassert_true((ret == 2), "strcspn failed");
+	ret = strcspn(str1, str3);
+	zassert_true((ret == 6), "strcspn not found str");
+}
+
+/**
+ *
+ * @brief test strncat function
+ *
+ */
+void test_strncat(void)
+{
+	char str[20] = "aa";
+	char src[20] = "bbccdd";
+
+	zassert_true(strncat(str, src, 2), "strncat failed");
+	zassert_true((strcmp(str, "aabb")) == 0, "strncat failed");
+}
+
+/**
+ *
+ * @brief test strrchr function
+ *
+ */
+void test_strrchr(void)
+{
+	char str[] = "abcccbf";
+	char *ptr;
+
+	zassert_is_null(strrchr(str, 'd'),
+		       "strrchr not found this word. failed");
+	ptr = strrchr(str, 'b');
+	zassert_true((strcmp(ptr, "bf") == 0), "strrchr failed");
+}
+
+/**
+ *
+ * @brief test strstr function
+ *
+ */
+void test_strstr(void)
+{
+	char *str1 = "aabbccdd";
+	char *str2 = "aa";
+	char *str3 = "ff";
+
+	zassert_is_null(strstr(str1, str3), "strstr aabbccdd with ff failed");
+	zassert_not_null(strstr(str1, str2), "strstr aabbccdd with aa failed");
+}
+
+/**
+ *
+ * @brief test strtol function
+ *
+ */
+void test_strtoxl(void)
+{
+	char buf1[20] = "10379aegi";
+	char buf2[20] = "10379aegi";
+	char buf3[20] = "010379aegi";
+	char buf4[20] = "0x10379aegi";
+	char *stop = NULL;
+	long ret;
+	unsigned long retul;
+
+	ret = strtol(buf3, &stop, 8);
+	zassert_true((ret == 543), "strtol base = 8 failed");
```

Comment:
Use `zassert_equal`

Reply:
@stephanosio why is this resolved? It's still using `zassert((ret == 543), msg);`?

Reply:
It's my mistake. I will fixing it.